pootle/apps/pootle_app/management/commands/update_external_tmserver.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+from hashlib import md5
+from optparse import make_option
+import os
+import sys
+
+from elasticsearch import helpers, Elasticsearch
+from translate.storage import factory
+
+# This must be run before importing Django.
+os.environ['DJANGO_SETTINGS_MODULE'] = 'pootle.settings'
+
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+
+
+BULK_CHUNK_SIZE = 5000
+
+
+class Command(BaseCommand):
+    help = "Add external translations to Local Translation Memory"
+    option_list = BaseCommand.option_list + (
+        make_option('--rebuild',
+                    action="store_true",
+                    dest='rebuild',
+                    default=False,
+                    help='Drop the entire TM on start and update everything '
+                         'from scratch'),
+        make_option('--dry-run',
+                    action="store_true",
+                    dest='dry_run',
+                    default=False,
+                    help='Report the number of translations to index and quit'),
+        make_option('--tm',
+                    action='store',
+                    dest='tm_server',
+                    help="TM to use. TM must exist on settings. TM will be "
+                         "created on the server if it doesn't exist"),
+        make_option('--target-language',
+                    action='store',
+                    dest='target_language',
+                    help="Target language to fallback to use in case it can't "
+                         "be guessed for any of the input files."),
+    )
+
+    def _parse_translations(self, *args, **options):
+        units = []
+
+        for filename in args:
+            store = factory.getobject(filename)
+            if (not store.gettargetlanguage() and
+                not self.target_language and
+                not options['dry_run']):
```

Comment:
IMHO when providing `--dry-run` we should not fail because it is is not possible to determine the target language, since we just want to report the number of translations to index.